### PR TITLE
[ExistentialLayout] Introduce getKind() and use it.

### PR DESCRIPTION
**Pull request by dcci**
Suggested by John McCall.

### CODE REVIEW

**Comment by dcci** [comment]
Sorry, wrong branch, will update in a second.

**Comment by dcci** [comment]
@swift-ci smoke test and merge

File: include/swift/AST/ExistentialLayout.h [hunk 1]
```diff
@@ -28,6 +28,8 @@ namespace swift {
   class ProtocolCompositionType;
 
 struct ExistentialLayout {
+  enum Kind { ClassExistential, ErrorExistential, OpaqueExistential };
```
Comment (rjmccall) [inline include/swift/AST/ExistentialLayout.h]: You could reasonably drop `Existential` from these names.

**Review by rjmccall — APPROVED** [review]
LGTM, but with a style suggestion if you're okay with waiting for another CI cycle.

File: include/swift/AST/ExistentialLayout.h (hunk 1 above)
Comment (dcci) [reply include/swift/AST/ExistentialLayout.h]: Sure, I can do that.

**Comment by dcci** [comment]
@swift-ci smoke test and merge

**Comment by dcci** [comment]
@swift-ci smoke test and merge

**Review by slavapestov — APPROVED** [review]
This is a great idea! Two optional suggestions below.

**Comment by dcci** [comment]
@swift-ci smoke test macos

**Comment by dcci** [comment]
@swift-ci please smoke test and merge

**Pull request merged**